src/lib.rs

```diff
@@ -1740,6 +1759,112 @@ impl LightningNode {
             .status)
     }
 
+    /// Check if clearing the wallet is feasible.
+    ///
+    /// If not feasible, it means the balance is either too high or too low for a reverse-swap to
+    /// be used.
+    pub fn is_clear_wallet_feasible(&self) -> Result<bool> {
```

Comment:
We could be more specific here and return `Result<ClearWalletSate>` or something such that we can actually tell whether the amount was too big or too small.

Once we have that separation, it would be nice to actually be informed about the min / max amount allowed. (I see this could be tricky because of the fees only being an estimation)

Reply:
It would be nice to provide that info to the user. I went with the current option because it's enough to support the implementation of the currently planned UI. Even if we change the UI later, should we already implement something that won't be initially used?

Reply:
Well, you're right, _premature optimization is the root of all evil_. However I'm surprised that we don't wont to expose this information in the planned UI.